pipeline-runner/R/gem2s-1-download_user_files.R

```diff
@@ -32,25 +31,65 @@ download_user_files <- function(input, pipeline_config, prev_out = list()) {
       fname <- basename(gem_key)
 
       # Preparing directories
-      local_fpath <- file.path(input_dir, sample, fname)
-      fs::dir_create(dirname(local_fpath))
+      local_fpath <- file.path(input_dir, sample_id, fname)
+
+      download_and_store(originals_bucket, gem_key, local_fpath, aws_config)
+    }
+  }
+}
 
-      # Download the file and store the output in a variable
-      c(body, ...rest) %<-% s3$get_object(
-        Bucket = pipeline_config$originals_bucket,
-        Key = gem_key
-      )
+file_names_v1 <- list(
+  barcodes10x = "barcodes.tsv.gz",
+  features10x = "features.tsv.gz",
+  matrix10x = "matrix.mtx.gz"
+)
 
-      # Write output to file
-      writeBin(body, con = local_fpath)
+file_types_by_technology <- list(
+  "10x" = list("barcodes10x", "features10x", "matrix10x")
+)
+
+get_gem2s_file_v2 <- function(project_id, sample_ids, s3_paths_by_sample, technology, originals_bucket, aws_config) {
```

Comment:
what is this doing? add documentation please

Reply:
It's like `get_gem2s_file_v1` only that instead of generating the s3 keys from the ids it fetches them from `s3_paths_by_sample`